x-pack/plugins/infra/public/components/asset_details/processes/processes.tsx

```diff
@@ -68,29 +84,33 @@ export const Processes = ({ currentTime, node, nodeType }: ProcessesProps) => {
     hostTerm,
     currentTime,
     sortBy,
-    parseSearchString(hostFlyoutOpen.searchFilter ?? '')
+    parseSearchString(hostFlyoutOpen?.searchFilter ?? searchText)
   );
 
-  const debouncedSearchOnChange = useMemo(
-    () =>
-      debounce<(queryText: string) => void>(
-        (queryText) => setHostFlyoutOpen({ searchFilter: queryText }),
-        500
-      ),
-    [setHostFlyoutOpen]
-  );
+  const debouncedSearchOnChange = useMemo(() => {
+    return debounce<(queryText: string) => void>(
+      (queryText) =>
+        setHostFlyoutOpen
+          ? setHostFlyoutOpen({ searchFilter: queryText })
+          : setSearchText(queryText),
```

Comment:
If I understood it correctly  `setHostFlyoutOpen` is called only when there is the need to store the state in the URL, right? 

I think after the `searchText` state initialization, we might want the component to only work internally with `searchText` state, and currently, there are a few places that check whether `hostFlyoutOpen.searchFilter` is undefined to decide what to use. 

So maybe the logic here should be something like this:

```ts
if (setHostFlyoutOpen) {
  setHostFlyoutOpen({ searchFilter: queryText })
}

setSearchText(queryText)

```

Here we'll always update the `searchText` state. Wdyt?

Reply:
Nice, I did that :)